tests/functional/test_chart_submitted_by_unauthorized_user.py

```diff
@@ -0,0 +1,294 @@
+
+    test_repo = TEST_REPO
+    repo = git.Repo()
+
+    # Differentiate between github runner env and local env
+    github_actions = os.environ.get("GITHUB_ACTIONS")
+    if github_actions:
+        # Create a new branch locally from detached HEAD
+        head_sha = repo.git.rev_parse('--short', 'HEAD')
+        local_branches = [h.name for h in repo.heads]
+        if head_sha not in local_branches:
+            repo.git.checkout('-b', f'{head_sha}')
+
+    current_branch = repo.active_branch.name
+    r = github_api(
+        'get', f'repos/{test_repo}/branches', bot_token)
+    branches = json.loads(r.text)
+    branch_names = [branch['name'] for branch in branches]
+    if current_branch not in branch_names:
+        logger.info(
+            f"{test_repo}:{current_branch} does not exists, creating with local branch")
+    repo.git.push(f'https://x-access-token:{bot_token}@github.com/{test_repo}',
+                  f'HEAD:refs/heads/{current_branch}', '-f')
+
+    base_branch = f'chart-tar-with-report-{current_branch}'
+    pr_branch = base_branch + '-pr'
+
+    secrets = Secret(test_repo, bot_name, bot_token, base_branch, pr_branch)
+    yield secrets
+
+    # Teardown step to cleanup branches
+    repo.git.worktree('prune')
+
+    if github_actions:
+        logger.info(f"Delete remote '{head_sha}' branch")
+        github_api(
+            'delete', f'repos/{secrets.test_repo}/git/refs/heads/{head_sha}', secrets.bot_token)
+
+    logger.info(f"Delete '{secrets.test_repo}:{secrets.base_branch}'")
+    github_api(
+        'delete', f'repos/{secrets.test_repo}/git/refs/heads/{secrets.base_branch}', secrets.bot_token)
+
+    logger.info(f"Delete '{secrets.test_repo}:{secrets.base_branch}-gh-pages'")
+    github_api(
+        'delete', f'repos/{secrets.test_repo}/git/refs/heads/{secrets.base_branch}-gh-pages', secrets.bot_token)
+
+    logger.info(f"Delete '{secrets.test_repo}:{secrets.pr_branch}'")
+    github_api(
+        'delete', f'repos/{secrets.test_repo}/git/refs/heads/{secrets.pr_branch}', secrets.bot_token)
+
+    logger.info(f"Delete local '{secrets.base_branch}'")
+    try:
+        repo.git.branch('-D', secrets.base_branch)
+    except git.exc.GitCommandError:
+        logger.info(f"Local '{secrets.base_branch}' does not exist")
+
+
+@scenario('features/chart_submitted_by_unauthorized_user.feature', "An unauthorized partner user submits a chart with report")
+def test_partner_chart_submission_by_unauthorized_user():
+    """An unauthorized partner user submits a chart with report"""
+
+@given("partner user is not present in the OWNERS file of the chart")
+def partner_user_not_present_in_the_chart_owners_file(secrets):
+    """partner user is not present in the OWNERS file of the chart"""
+    secrets.vendor_type = 'partners'
+    secrets.vendor = get_unique_vendor('hashicorp')
+
+@given("the user creates a branch to add a new chart version")
+def the_user_has_created_a_error_free_chart_tar_with_report(secrets):
```

Comment:
any reason not to match the function name withe the when description. Same applies to other methods.